pkg/aws/role/role.go

```diff
@@ -26,14 +26,12 @@ import (
 )
 
 // Switch func uses credentials API to automatically generates New Credentials for a given role.
-func Switch(ctx context.Context, accessKeyID string, secretAccessKey string, role string, duration time.Duration) (*credentials.Credentials, error) {
-	creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, "")
-	sess, err := session.NewSession(aws.NewConfig().WithCredentials(creds))
+func Switch(ctx context.Context, creds *credentials.Credentials, role string, duration time.Duration) (*credentials.Credentials, error) {
```

Comment:
We should be able to unexport `Switch` going forward. Callers should be using the `aws.GetCredentials` method to get credentials and it will take care of switching roles if required.

Reply:
See `objectstore_test.go` for example